src/pybullet_industrial/gcode_class.py

```diff
@@ -0,0 +1,455 @@
+        G-Command it either runs a linear or circular interpolation and calls
+        the move-methods depending if a tool is activated or not.
+
+        Args:
+            g_com: G-Command from the g_code File
+            r_val: Radius for the circular interpolation
+
+        Returns:
+            None
+        """
+        orientation = p.getQuaternionFromEuler(self.new_or)
+
+        # Building the Path if there is a linear G1 interpolation
+        if g_com == 1:
+            path = pi.linear_interpolation(self.last_point,
+                                           self.new_point,
+                                           self.interpolation_steps)
+
+        # Building the path if there is a circular interpolation
+        elif g_com in [2, 3]:
+            if g_com == 2:
+                path = pi.circular_interpolation(self.last_point,
+                                                 self.new_point, r_val,
+                                                 self.interpolation_steps,
+                                                 self.plane, True)
+            else:
+                path = pi.circular_interpolation(self.last_point,
+                                                 self.new_point, r_val,
+                                                 self.interpolation_steps,
+                                                 self.plane, False)
+
+        path.orientations = np.transpose([orientation]
+                                         * len(path.orientations[0]))
+
+        # Moving endeffector or the robot
+        if not self.active_endeffector == -1:
+            self.move_along_path(path)
+        else:
+            self.move_robot(path)
+
+    def move_along_path(self, path: pi.ToolPath):
+        """Moves the active endeffector along a designated Path.
+
+        Args:
+            path(pi.ToolPath): Array of points defining the path
+        Returns:
+            None
+        """
+        e = self.active_endeffector
+
+        for positions, orientations, tool_path in path:
+
+            for _ in range(10):
+
+                self.endeffector_list[e].set_tool_pose(positions, orientations)
+                for _ in range(10):
+                    p.stepSimulation()
+                    time.sleep(self.sleep)
+
+                current_position = self.endeffector_list[e].get_tool_pose()[0]
+                current_orientation = np.array(p.getEulerFromQuaternion(
+                    self.endeffector_list[e].get_tool_pose()[1]))
+
+                position_error = np.linalg.norm(current_position-positions)
+                orientations_euler = np.array(
+                    p.getEulerFromQuaternion(orientations))
+                orientation_error = np.linalg.norm(current_orientation -
+                                                   orientations_euler)
+
+                if position_error < 0.02 and orientation_error < 0.004:
+                    break
+
+    def move_robot(self, path: pi.ToolPath):
```

Comment:
almost a duplicate of the function above. Should be shortened to only include what is different to `move_along_path`